### PR TITLE
Feat: limit field length

### LINKED ISSUE #291: Manage overflow for parameters
Limit parameters to the following:

studio/reflection: 9999
name/phone/address/email: 160 char? 
exam: 12 chars
maxScore: 10 digits
tag: 160 char

### PR DESCRIPTION
**Pull request by Zer0Legion**
fixes #291

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Email.java [hunk 1]
```diff
@@ -12,22 +12,24 @@ public class Email {
...
             + "separated by periods.\n"
             + "The domain name must:\n"
             + "    - end with a domain label at least 2 characters long\n"
             + "    - have each domain label start and end with alphanumeric characters\n"
             + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.";
     // alphanumeric and special characters
-    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
-    private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
-            + ALPHANUMERIC_NO_UNDERSCORE + ")*";
+    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]{1,80}";
```
Comment (jayllo-c) [inline src/main/java/seedu/address/model/person/Email.java:27]: Good job adding the limit to the max number of emails

File: src/main/java/seedu/address/model/person/Name.java [hunk 2]
```diff
@@ -10,13 +10,14 @@
...
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters and spaces up to 160 characters, "
+                    + "and it should not be blank";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]{1,159}";
```
Comment (jayllo-c) [inline src/main/java/seedu/address/model/person/Name.java:20]: Realistic in setting the max numbet of characters in a name to 160 with consideration to outlier names which may be quite long.

File: src/main/java/seedu/address/model/person/Phone.java [hunk 3]
```diff
@@ -11,8 +11,8 @@ public class Phone {
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+            "Phone numbers should only contain numbers, and it should be between 3 and 160 digits long inclusive.";
+    public static final String VALIDATION_REGEX = "\\d{3,160}";
```
Comment (jayllo-c) [inline src/main/java/seedu/address/model/person/Phone.java:15]: Might wanna lower the max number of numbers from 160 to 30?

**Review by jayllo-c — COMMENTED** [review]
Apart from phone numbers, theimit placed on other fields are realistic and will help prevent text from extending beyond the UI limits

File: src/test/java/seedu/address/model/exam/ExamTest.java [hunk 4]
```diff
@@ -13,7 +13,6 @@ public class ExamTest {
     @Test
     public void isValidName() {
         assertTrue(Exam.isValidExamName("Midterm")); // alphanumeric characters
-        assertTrue(Exam.isValidExamName("Midterm Exam")); // alphanumeric characters with spaces
```
Comment (delishad21) [inline src/test/java/seedu/address/model/exam/ExamTest.java:16]: Why was this test removed?

File: src/main/resources/view/ExamListCard.fxml [hunk 5]
```diff
@@ -28,7 +27,7 @@
         </Label>
         <Label fx:id="name" styleClass="cell_big_label" text="\$exam_name" />
       </HBox>
-         <Label text="Max Score: " styleClass="cell_small_label"/>
+         <Label styleClass="cell_small_label" text="Max Score: " />
```
Comment (delishad21) [inline src/main/resources/view/ExamListCard.fxml:30]: Hmm this is an interesting swap in order. I'm assuming Scene Builder did this

File: src/main/java/seedu/address/model/person/Score.java [hunk 6]
```diff
@@ -12,13 +12,13 @@
...
-            + "Scores can have up to 2 decimal places.";
+            "Scores should be numeric, and it should have between 1 and 7 digits inclusive,"
+            + "with up to 2 decimal places.";
 
     /*
-     * The score must be a non-negative integer. and only have
+     * The score must be a non-negative integer. and only have up to 7 digits and 2 decimal places.
      */
-    public static final String VALIDATION_REGEX = "^\\d+(\\.\\d{1,2})?$";
+    public static final String VALIDATION_REGEX = "^\\d{1,7}(\\.\\d{1,2})?$";
```
Comment (delishad21) [inline src/main/java/seedu/address/model/person/Score.java:21]: Realistic limit set on double. good also as beyond 7 digits, string conversion for double starts to get a little wonky,

**Review by delishad21 — COMMENTED** [review]
Realistic character limits set for all our fields. Will help to prevent bugs and prevent antagonistic testers from trying to break our program

File: src/main/resources/view/ExamListCard.fxml (hunk 5 above)
Comment (Zer0Legion) [reply src/main/resources/view/ExamListCard.fxml:30]: That is indeed interesting. Looks like SceneBuilder is insightfully trying to parallel life, where it does not place maximum scores as its first priority, as we all should.

File: src/test/java/seedu/address/model/exam/ExamTest.java (hunk 4 above)
Comment (Zer0Legion) [reply src/test/java/seedu/address/model/exam/ExamTest.java]: Apologies, I have added it back in.

**Pull request merged**